lib/spack/spack/cmd/ci.py

```diff
@@ -580,14 +585,14 @@ def ci_rebuild(args):
 
     # ["x", "y"] -> "'x' 'y'"
     args_to_string = lambda args: " ".join("'{}'".format(arg) for arg in args)
-
+    export_command = "set" if sys.platform == "win32" else "export"
     commands = [
         # apparently there's a race when spack bootstraps? do it up front once
-        [SPACK_COMMAND, "-e", env.path, "bootstrap", "now"],
+        [SPACK_COMMAND, "-e", encode_path(env.path), "bootstrap", "now"],
         [
```

Comment:
We can just remove all of the makefile parts here.

Reply:
Works for me!

Reply:
It looks like that hasn't happened yet (still see makefile references below). This could be replaced by a `spack install`, although I don't know why it was done with `depfile` here before so there could be some performance differences

Reply:
I haven't taken out the makefile stuff yet as I'm waiting to see if #39529 would progress (Harmen wrote this initially, and that PR of his also removes this, given that I think he's the best person to take this out), worst case I cherry-pick the relevant parts of that PR over to this one.

Reply:
The `depfile` was used to improve parallel installation of dependencies, but as I understand it, that did not end up being the location of the performance bottleneck the `depfile` was intended to improve, so we can refactor away from  `depfile` approach.

Reply:
What confused me in this PR is that this includes changes to a function that still cannot run (on Windows) given the changes in the PR: IMO the changes in this file should move to another PR.